src/crypto/internal/boring/goboringcrypto.h

```diff
@@ -19,218 +19,687 @@
-int _goboringcrypto_HMAC_Final(GO_HMAC_CTX*, uint8_t*, unsigned int*);
-size_t _goboringcrypto_HMAC_size(const GO_HMAC_CTX*);
+static inline const GO_EVP_MD*
+_goboringcrypto_EVP_sha1(void)
+{
+	return EVP_sha1();
+}
+
+static inline const GO_EVP_MD*
+_goboringcrypto_EVP_sha224(void)
+{
+	return EVP_sha224();
+}
+
+static inline const GO_EVP_MD*
+_goboringcrypto_EVP_sha256(void)
+{
+	return EVP_sha256();
+}
+
+static inline const GO_EVP_MD*
+_goboringcrypto_EVP_sha384(void)
+{
+	return EVP_sha384();
+}
+
+static inline const GO_EVP_MD*
+_goboringcrypto_EVP_sha512(void)
+{
+	return EVP_sha512();
+}
+
+static inline int
+_goboringcrypto_EVP_MD_type(const GO_EVP_MD* arg0)
+{
+	return EVP_MD_type(arg0);
+}
+
+static inline size_t
+_goboringcrypto_EVP_MD_size(const GO_EVP_MD* arg0)
+{
+	return EVP_MD_size(arg0);
+}
+
+
+#include <openssl/hmac.h>
+
+typedef HMAC_CTX GO_HMAC_CTX;
+static inline void
+_goboringcrypto_HMAC_CTX_init(GO_HMAC_CTX* arg0)
+{
+	HMAC_CTX_init(arg0);
+}
+
+static inline void
+_goboringcrypto_HMAC_CTX_cleanup(GO_HMAC_CTX* arg0)
+{
+	HMAC_CTX_cleanup(arg0);
+}
+
+static inline int
+_goboringcrypto_HMAC_Init(GO_HMAC_CTX* arg0, const void* arg1, int arg2, const GO_EVP_MD* arg3)
+{
+	return HMAC_Init(arg0, arg1, arg2, arg3);
+}
+
+static inline int
+_goboringcrypto_HMAC_Update(GO_HMAC_CTX* arg0, const uint8_t* arg1, size_t arg2)
+{
+	return HMAC_Update(arg0, arg1, arg2);
+}
+
+static inline int
+_goboringcrypto_HMAC_Final(GO_HMAC_CTX* arg0, uint8_t* arg1, unsigned int* arg2)
+{
+	return HMAC_Final(arg0, arg1, arg2);
+}
+
+static inline size_t
+_goboringcrypto_HMAC_size(const GO_HMAC_CTX* arg0)
+{
+	return HMAC_size(arg0);
+}
+
 int _goboringcrypto_HMAC_CTX_copy_ex(GO_HMAC_CTX *dest, const GO_HMAC_CTX *src);
 
-// #include <openssl/aes.h>
-typedef struct GO_AES_KEY { char data[244]; } GO_AES_KEY;
-int _goboringcrypto_AES_set_encrypt_key(const uint8_t*, unsigned int, GO_AES_KEY*);
-int _goboringcrypto_AES_set_decrypt_key(const uint8_t*, unsigned int, GO_AES_KEY*);
-void _goboringcrypto_AES_encrypt(const uint8_t*, uint8_t*, const GO_AES_KEY*);
-void _goboringcrypto_AES_decrypt(const uint8_t*, uint8_t*, const GO_AES_KEY*);
-void _goboringcrypto_AES_ctr128_encrypt(const uint8_t*, uint8_t*, size_t, const GO_AES_KEY*, uint8_t*, uint8_t*, unsigned int*);
+#include <openssl/aes.h>
+
+typedef AES_KEY GO_AES_KEY;
+static inline int
+_goboringcrypto_AES_set_encrypt_key(const uint8_t* arg0, unsigned int arg1, GO_AES_KEY* arg2)
+{
+	return AES_set_encrypt_key(arg0, arg1, arg2);
+}
+
+static inline int
+_goboringcrypto_AES_set_decrypt_key(const uint8_t* arg0, unsigned int arg1, GO_AES_KEY* arg2)
+{
+	return AES_set_decrypt_key(arg0, arg1, arg2);
+}
+
+static inline void
+_goboringcrypto_AES_encrypt(const uint8_t* arg0, uint8_t* arg1, const GO_AES_KEY* arg2)
+{
+	AES_encrypt(arg0, arg1, arg2);
+}
+
+static inline void
+_goboringcrypto_AES_decrypt(const uint8_t* arg0, uint8_t* arg1, const GO_AES_KEY* arg2)
+{
+	AES_decrypt(arg0, arg1, arg2);
+}
```

Comment:
Are you still using these raw AES functions anywhere? Perhaps they should be dropped?

Reply:
removed